src/custom-element.js

```diff
@@ -1251,8 +1258,21 @@ export function getElementServiceIfAvailable(win, id, providedByElement) {
   if (s) {
     return s;
   }
-  if (!isElementScheduled(win, providedByElement)) {
-    return Promise.resolve(null);
-  }
-  return getElementService(win, id, providedByElement);
+  // Microtask is necessary to ensure that window.ampExtendedElements has been
+  // initialized.
+  return Promise.resolve().then(() => {
+    if (isElementScheduled(win, providedByElement)) {
+      return getServicePromise(win, id);
+    }
+    // Wait for HEAD to fully form before denying access to the service.
+    if (!win.document.body) {
```

Comment:
There could be a raise condition here: Body is now available, but element discovery has not run again.
Better to just always do this?